gr-blocks/lib/tuntap_pdu_impl.cc

```diff
@@ -168,9 +168,12 @@ namespace gr {
       ifr.ifr_mtu = MTU;
 
       /* try to set MTU */
-      if ((err = ioctl(sfd, SIOCSIFMTU, (void *) &ifr)) < 0)
-        return err;
+      if ((err = ioctl(sfd, SIOCSIFMTU, (void *) &ifr)) < 0) {
```

Comment:
Couldn't we just do 
<pre>
err = ioctl(sfd, SIOCSIFMTU, (void *) &ifr);
close(sfd);
return (err < 0) ? err : MTU;
</pre>

Reply:
also, and that's what I'm actually worried about: Shouldn't we be returning `errno` instead of `err` (which will always be -1 in case of error)?